docs/current_docs/api/fs-filters.mdx

```diff
@@ -0,0 +1,308 @@
+
+- Repositories containing binary objects. Binary objects, such as audio/video files, are usually extremely large in size and therefore take time to upload. If they're not directly relevant, you'll usually want your Dagger Function to ignore them.
+
+  :::tip
+  The `.git` directory is a good example of both these cases. It contains a lot of data, including large binary objects, and for projects with a long version history, it can sometimes be larger than your actual source code.
+  :::
+
+- Dependencies. If you're developing locally, you'll typically have your project dependencies installed locally as well: `node_modules` (Node.js), `.venv` (Python), `vendor` (PHP) and so on. When you call your Dagger Function locally, Dagger will upload all these installed dependencies as well. This is both bad practice and inefficient. Typically, you'll want your Dagger Function to ignore locally-installed dependencies and only operate on the project source code.
+
+:::note
+At the time of writing, Dagger does not read exclusion patterns from existing `.dockerignore`/`.gitignore` files. If you already use these files, you'll need to manually implement the same patterns in your Dagger Function.
+:::
+
+To implement a pre-call filter in your Dagger Function, add an `ignore` parameter to your `Directory` argument. The `ignore` parameter follows the `.gitignore` syntax: prefixing a path with `!` will actually include it instead of ignoring it.
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+Here's an example of a Dagger Function that excludes everything in a given directory except Go source code files:
+
+```go file=./snippets/fs-filters/pre-call/go/main.go
+```
+
+</TabItem>
+<TabItem value="Python">
+
+Here's an example of a Dagger Function that excludes everything in a given directory except Python source code files:
+
+```python file=./snippets/fs-filters/pre-call/python/main.py
+```
+
+</TabItem>
+<TabItem value="TypeScript">
+
+Here's an example of a Dagger Function that excludes everything in a given directory except TypeScript source code files:
+
+```typescript file=./snippets/fs-filters/pre-call/typescript/index.ts
+```
+
+</TabItem>
+</Tabs>
+
+Here are a few examples of useful patterns:
+
+<Tabs groupId="language">
+<TabItem value="Go">
+```go
+// exclude Go tests and test data
++ignore=["**_test.go", "**/testdata/**"]
+
+// exclude binaries
++ignore=["bin"]
+
+// exclude Python dependencies
++ignore=["**/.venv","**/__pycache__"]
+
+// exclude Node.js dependencies
++ignore=["**/node_modules"]
+
+// exclude Git metadata
++ignore=["*", "!.git", "!**/.gitignore"]
+```
+</TabItem>
+<TabItem value="Python">
+```python
+# exclude Go tests and test data
+Ignore(["**_test.go", "**/testdata/**"])
+
+# exclude binaries
+Ignore(["bin"])
+
+# exclude Python dependencies
+Ignore(["**/.venv","**/__pycache__"])
+
+# exclude Node.js dependencies
+Ignore(["**/node_modules"])
+
+# exclude Git metadata
+Ignore(["*", "!.git", "!**/.gitignore"])
```

Comment:
Here you actually exclude everything but include Git metadata.

Reply:
Updated, see if correct now?